pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go

```diff
@@ -126,6 +147,39 @@ type klusterletAgent struct {
 	getSSARFunc    getSelfSubjectAccessReviewsFunc
 }
 
+// Check both agent deployments, if the desired replicas is equal to available replicas, return available condition
+func checkAgentsDeployment(ctx context.Context, kubeClient kubernetes.Interface, agents []klusterletAgent) metav1.Condition {
+	availableMessages := []string{}
+	for _, agent := range agents {
+		deployment, err := kubeClient.AppsV1().Deployments(agent.namespace).Get(ctx, agent.deploymentName, metav1.GetOptions{})
+		if err != nil {
+			return metav1.Condition{
+				Type:    klusterletProgressing,
+				Status:  metav1.ConditionTrue,
+				Reason:  "GetDeploymentFailed",
+				Message: fmt.Sprintf("Failed to get deployment %q %q: %v", agent.namespace, agent.deploymentName, err),
+			}
+		}
+		if unavailablePod := helpers.NumOfUnavailablePod(deployment); unavailablePod > 0 {
+			return metav1.Condition{
```

Comment:
I think you would want to also return available=true also.

Reply:
OK, if the deployment is partially ready, we can also set the klusterlet condition as `Available`, let me update the code then.

Reply:
> I think you would want to also return available=true also.

Hi @qiujian16 ,
Per your comment above
```
progressing is a status that deployment is not fully up, or being rolled out.
```

Here, when `unavailablePod > 0`, it means the deployment is not fully up, and should be set to condition `Progressing`, right?